### PR TITLE
Intra-series navigation

### LINKED ISSUE #1828: Add next and previous buttons to the exercise page
On an exercise page within a series (if the series navigation at the right is active), add a next and previous button to exercise card header. If an exercise is the first or last in a series, replace the useless button with a "return to course".

image

### PR DESCRIPTION
**Pull request by thepieterdc**
This pull request adds the possibility to navigate between exercises of the same series using chevrons in the header.

image

Closes #1828.

### CODE REVIEW

**Comment by beardhatcode** [comment]
Could you make the buttons slightly less prominent (`rgba(255, 255, 255, 0.7)`)
Screenshot_2020-04-07 ISBN-MD

Are these buttons realy that usefull if we also have them in the right sidebar?

**Comment by pdawyndt** [comment]
@beardhatcode yes we already have a right sidebar with exercises in a series!

**Comment by beardhatcode** [comment]
The last `>` now goes to the cource overview page. Would it be possible to make it jump to te correct series as well?

**Comment by pdawyndt** [comment]
Same for first `<`

**Comment by thepieterdc** [comment]
> The last `>` now goes to the cource overview page. Would it be possible to make it jump to the correct series as well?

> Same for first `<`

This was indeed intended by @bmesuere, I am currently implementing this.

File: app/assets/stylesheets/models/exercises.css.less [hunk 1]
```diff
@@ -23,6 +23,28 @@
...
+      color: inherit;
+      display: flex;
+      font-size: 24px;
+      font-weight: 300;
+      line-height: normal;
+      overflow: hidden;
+      margin-left: auto;
+
+      a {
+        color: #ffffff;
```
Comment (charvp) [inline app/assets/stylesheets/models/exercises.css.less:41]: Use `@header-fg` here.

**Comment by pdawyndt** [comment]
Wouldn't it be more natural to jump to the first exercise of the next series or the last exercise of previous series, instead of going back to the course page? This way it would be easier to trace back to where you came from, if you unintentionally move beyond the limits of the series.

**Comment by thepieterdc** [comment]
> Wouldn't it be more natural to jump to the first exercise of the next series or the last exercise of previous series, instead of going back to the course page? This way it would be easier to trace back to where you came from, if you unintentionally move beyond the limits of the series.

That was indeed intended

**Comment by thepieterdc** [comment]
> Wouldn't it be more natural to jump to the first exercise of the next series or the last exercise of previous series, instead of going back to the course page? This way it would be easier to trace back to where you came from, if you unintentionally move beyond the limits of the series.

@bmesuere opted not to implement this, since you can simply use the back-button of your browser to undo this action.

I will be adding some tests for the newly added helper methods later today.

**Comment by thepieterdc** [comment]
I have implemented your suggestions (colours on screenshot are due to image compression).

image

File: app/assets/stylesheets/components/btn.css.less [hunk 2]
```diff
@@ -189,6 +189,18 @@
   height: 33px;
 }
 
+.btn-icon-inverted {
+  &:extend(.btn-icon);
+
+  box-shadow: none;
```
Comment (bmesuere) [inline app/assets/stylesheets/components/btn.css.less:195]: can be removed

File: app/assets/stylesheets/components/btn.css.less [hunk 3]
```diff
@@ -189,6 +189,18 @@
...
+  color: @white-secondary;
+
+  &:hover {
+    box-shadow: none;
```
Comment (bmesuere) [inline app/assets/stylesheets/components/btn.css.less:199]: can be removed

File: app/assets/stylesheets/models/exercises.css.less [hunk 4]
```diff
@@ -23,6 +23,24 @@
 }
 
 // exercise
+.exercise-description {
+  .card-title {
+    display: flex;
+    flex-direction: row;
+
+    .card-title-navigation {
+      color: inherit;
```
Comment (bmesuere) [inline app/assets/stylesheets/models/exercises.css.less:32]: can be removed

File: app/views/exercises/show.html.erb [hunk 5]
```diff
@@ -16,6 +16,23 @@
...
+        <% if @series.present? %>
+          <%
+            previous_exercise, next_exercise = previous_next_exercise_path(@series, @exercise)
+            previous_link = previous_exercise || breadcrumb_series_path(@series, current_user)
+            previous_tooltip = previous_exercise ? t('.navigation.previous') : t('.navigation.back_to_course')
+            next_link = next_exercise || breadcrumb_series_path(@series, current_user)
+            next_tooltip = next_exercise ? t('.navigation.next') : t('.navigation.back_to_course')
+          %>
+          <div class="card-title-navigation hidden-print">
+            <%= link_to previous_link, class: 'btn btn-icon-inverted', title: previous_tooltip, data: {toggle: 'tooltip', placement: 'top'} do %>
```
Comment (bmesuere) [inline app/views/exercises/show.html.erb:28]:
```suggestion
            <%= link_to previous_link, class: 'btn-icon-inverted', title: previous_tooltip, data: {toggle: 'tooltip', placement: 'top'} do %>
```

File: app/views/exercises/show.html.erb [hunk 6]
```diff
@@ -16,6 +16,23 @@
...
+              <i class="mdi mdi-chevron-left"></i>
+            <% end %>
+            <%= link_to next_link, class: 'btn btn-icon-inverted', title: next_tooltip, data: {toggle: 'tooltip', placement: 'top'} do %>
```
Comment (bmesuere) [inline app/views/exercises/show.html.erb:31]:
```suggestion
            <%= link_to next_link, class: 'btn-icon-inverted', title: next_tooltip, data: {toggle: 'tooltip', placement: 'top'} do %>
```

File: config/locales/views/exercises/en.yml [hunk 7]
```diff
@@ -57,6 +57,10 @@ en:
       alert_removed: This exercise is no longer available, but can still see your old submissions.
       deadline_passed: "The deadline for this exercise was at %{deadline}. You can still submit, but your submissions may not be accounted for."
       deadline_in_5_minutes: "The deadline for this exercise is at %{deadline}, which is less than 5 minutes away."
+      navigation:
+        back_to_course: Back to course overview
+        previous: Previous exercise in series
```
Comment (bmesuere) [inline config/locales/views/exercises/en.yml:62]:
```suggestion
        previous: Previous exercise
```

File: config/locales/views/exercises/en.yml [hunk 8]
```diff
@@ -57,6 +57,10 @@ en:
...
+        next: Next exercise in series
```
Comment (bmesuere) [inline config/locales/views/exercises/en.yml:63]:
```suggestion
        next: Next exercise
```

File: config/locales/views/exercises/nl.yml [hunk 9]
```diff
@@ -57,6 +57,10 @@ nl:
       alert_removed: Deze oefening is niet meer beschikbaar, maar je kan nog steeds je oude oplossingen bekijken.
       deadline_passed: "De deadline voor deze oefening was om %{deadline}. Je kan nog steeds indienen, maar de lesgever houdt er mogelijk geen rekening meer mee."
       deadline_in_5_minutes: "De deadline voor deze oefening is om %{deadline}, wat over minder dan 5 minuten is."
+      navigation:
+        back_to_course: Terug naar cursusoverzicht
+        previous: Vorige oefening in deze reeks
```
Comment (bmesuere) [inline config/locales/views/exercises/nl.yml:62]:
```suggestion
        previous: Vorige oefening
```

File: config/locales/views/exercises/nl.yml [hunk 10]
```diff
@@ -57,6 +57,10 @@ nl:
...
+        next: Volgende oefening in deze reeks
```
Comment (bmesuere) [inline config/locales/views/exercises/nl.yml:63]:
```suggestion
        next: Volgende oefening
```

**Pull request merged**